PF/Assignment1/Booking.java

```diff
@@ -0,0 +1,293 @@
+
+					}
+					if (count == 4) {
+						price = price + c;
+
+					}
+				} else {
+					count++;
+					if (count == 6) {
+						count = 0;
+
+						int num = Integer.parseInt(number);
+						if (num == 1) {
+							PassengerTrain passenger = new PassengerTrain(
+									Integer.parseInt(trainNum), source,
+									destination, Integer.parseInt(price),
+									Integer.parseInt(availableSeat));
+							passengerList.add(passenger);
+						} else if (num == 2) {
+							GoodsTrain goods = new GoodsTrain(
+									Integer.parseInt(trainNum), source,
+									destination, Integer.parseInt(price),
+									Double.parseDouble(availableSeat));
+							goodsList.add(goods);
+						}
+						number = "";
+						source = "";
+						destination = "";
+						availableSeat = "";
+						price = "";
+						trainNum = "";
+					}
+				}
+
+			}
+			fr.close();
+
+			do {
+				System.out.println("Enter your name");
+				sc = new Scanner(System.in);
+				String userName = sc.next();
+				int temp1 = 0;
+				System.out.println("What type of train you want");
+				System.out.println("1.Passenger Train");
+				System.out.println("2.Goods Train");
+				int ch = sc.nextInt();
+				switch (ch) {
+				case 1:
+					System.out.println("---CURRENT TRAVELLING CHART");
+					System.out
+							.println("TRAIN No STARTINGPOINT ENDPOINT	PRICE SEATSAVAILABLE");
+					Iterator<PassengerTrain> itr3 = passengerList.iterator();
+					while (itr3.hasNext()) {
+						PassengerTrain object5 = itr3.next();
+						System.out.println(object5.trainNumber + "		"
+								+ object5.startingPoint + "	"
+								+ object5.endPoint + "	" + object5.price + "	"
+								+ object5.seats);
+
+					}
+
+					System.out.println("Enter from where you want to go");
+					String fromPlace = sc.next();
+					System.out.println("Enter where you want to go");
+					String toPlace = sc.next();
+					Iterator<PassengerTrain> itr = passengerList.iterator();
+					while (itr.hasNext()) {
+						PassengerTrain object1 = itr.next();
+						if ((object1.startingPoint.equalsIgnoreCase(fromPlace))
+								&& (object1.endPoint.equalsIgnoreCase(toPlace))) {
+							System.out.println("Enter no of seats");
+							int seats = sc.nextInt();
+							if (seats < object1.seats) {
+								double amount = seats * object1.price;
+								System.out.println("enter payment mode");
+								System.out.println("1.CreditCard");
+								System.out.println("2.Wallet");
+								System.out.println("3.Netbanking");
+								int choose = sc.nextInt();
+								PaymentMode payment = new PaymentMode();
+								switch (choose) {
+								case 1:
+
+									payment.creditCard(amount);
+									break;
+
+								case 2:
+
+									payment.wallet(amount);
+									break;
+
+								case 3:
+									payment.netBanking(amount);
+									break;
+
+								default:
+									System.out.println("Enter right values");
+									break;
+
+								}
+								int newSeats = object1.seats - seats;
```

Comment:
ACTION: Why you created new int variable here? you do this directly object1.setSeats(object1.seats - seats);